numpy/core/src/multiarray/mapping.c

```diff
@@ -2143,6 +2144,7 @@ array_assign_subscript(PyArrayObject *self, PyObject *ind, PyObject *op)
     return -1;
 
   success:
+    PyArray_ResolveWritebackIfCopy(view);
```

Comment:
Is this related to `np.nditer`?

Reply:
Sorry if this is a naive question, I'm just coming back to this:

Why is this line needed? Doesn't the context manager `exit` take care of any needed writebacks? Furthermore, let's say that within the context manager the user creates a view of the array they got from nditer, and writes to the view (Thus calling `Writeback`) and then the context manager exits, doesn't that double-call the writeback code?

This PR segfaults on the following for me:
```python
a = np.arange(24, dtype='f8').reshape(2, 3, 4).T
with np.nditer(a, [],[['readwrite', 'updateifcopy']],casting='same_kind',op_dtypes=[np.dtype('f4')]) as i:
    i.operands[0][...] = 3
    i.operands[0][...] = 14
```
which I suspect is related to my comment, but I haven't checked carefully.

Reply:
good catch. Added your test and fixed in c6a1a2c6d09. The context manager does indeed clean up, but this function can possibly allocate a new ndarray with ``writebackifcopy`` semantics that need to be cleaned up. The fix was to call the resolve function iff a new ``view`` is created